packages/common/src/interfaces/compositeEditorOpenDetailOption.interface.ts

```diff
@@ -104,6 +104,12 @@ export interface CompositeEditorOpenDetailOption {
    */
   onClose?: () => Promise<boolean>;
 
+  /**
+   * onDispose callback allows user to execute something when the modal is disposed (closed and removed from the DOM).
+   * This will be called regardless of whether there are changes done in the form or not.
+   */
+  onDispose?: () => void;
```

Comment:
~why create a new event, or mainly why is `onClose` event not good enough? I would prefer to have less code... is it possible to update the `onClose` instead to make it more versatile?~

ahhhh nevermind, I just read the docs I wrote a while ago just above and yes now it makes sense but I wonder if maybe that `onClose` should be renamed to a more representative name in the future (that would be a breaking change though)

> For example, when the user opens the modal window and start changing a few inputs in the form but then decides to the form, this is when the `onClose` gets executed (and to be clear, it only gets executed when there's changes in the form and a close action is clicked, it won't execute when there's no changes).

Reply:
I agree the onClose is a bit tricky. Wouldn't go for the breaking change though.